students/eowyn/session11/iterator_1.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+
+"""
+Simple iterator examples
+"""
+
+
+class IterateMe_1(object):
+    """
+    About as simple an iterator as you can get:
+
+    returns the sequence of numbers from zero to 4
+    ( like range(4) )
+    """
+    def __init__(self, stop=5):
+        self.current = -1
+        self.stop = stop
+    def __iter__(self):
+        return self
+    def __next__(self):
+        self.current += 1
+        if self.current < self.stop:
+            return self.current
+        else:
+            raise StopIteration
+
+
+class IterateMe_2():
+    def __init__(self, start=0, stop=5, step=1):
+        self.current = start-step
+        self.stop = stop
+        self.step = step
+        self.start = start
+    def __iter__(self):
+        return IterateMe_2(self.start,self.stop,self.step)
```

Comment:
you usually want to return self -- that is, this same object.

you can reset self.current so it restarts.